uber/site_sections/preregistration.py

```diff
@@ -524,3 +527,22 @@ def shirt_reorder(self, session, message='', **params):
             'message': message,
             'attendee': attendee
         }
+
+    def apply_promo_code(self, session, message='', **params):
+        if 'id' and 'code' in params:
+            if params['code'] is not '':
+                attendee, group = self._get_unsaved(params['id'])
+                if not attendee:
+                    message = 'Attendee Not Found'
+                else:
```

Comment:
There is a lot of nesting here. In the rest of the code, we:
1. Check for various problems and use them to set the `message` variable
2. Check if the `message` variable exists, which implies failure
3. Do other things if the message is not set (sometimes, including setting the message to a 'success' message)

This helps avoid this kind of nesting, making the code easier to read.